webserver/pkgpkr/static/main.js

```diff
@@ -0,0 +1,42 @@
+$(document).ready( function () {
+    var table = $('.display-data-tables').DataTable({
+        "bLengthChange": false,
+        dom: 'itp'
+    });
+
+    $('#categoryName').keyup( function() {
+        table.column(1).search($(this).val()).draw();
+    });
+
+    $('#recommendationFilter').keyup( function() {
+        table.search($(this).val()).draw();
+    });
+});
+
+function categoryClick(value) {
+    var categoryName = document.getElementById('categoryName');
+    document.getElementById('categoryName').value = value;
+    triggerEvent(categoryName, 'keyup');
+    document.getElementById('categoryClear').style.visibility = "visible";
+}
+
+function triggerEvent(el, type){
+ if ('createEvent' in document) {
+      // modern browsers, IE9+
+      var e = document.createEvent('HTMLEvents');
+      e.initEvent(type, false, true);
+      el.dispatchEvent(e);
+  } else {
+      // IE 8
```

Comment:
I think it's safe to ignore IE8. It's about 0.3% of all browsers, not worth the extra code complexity imo.

Reply:
Got it, deleted else part